rotkehlchen/chain/evm/decoding/monerium/decoder.py

```diff
@@ -0,0 +1,99 @@
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+class MoneriumCommonDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            native_asset: Asset,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.native_asset = native_asset
+        self.eure_token = native_asset.resolve_to_evm_token()
+
+    def _decode_mint_and_burn(self, context: DecoderContext) -> DecodingOutput:
+        """Decode mint and burn events for monerium"""
+
+        if context.tx_log.topics[0] != ERC20_OR_ERC721_TRANSFER:
+            return DEFAULT_DECODING_OUTPUT
+
+        from_address = hex_or_bytes_to_address(context.all_logs[0].topics[1])
+        to_address = hex_or_bytes_to_address(context.all_logs[0].topics[2])
+
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data)
+        amount = token_normalized_value_decimals(amount_raw, self.eure_token.decimals)
+
+        event = None
+        if from_address == ZERO_ADDRESS:
+            # Create a mint event
+            mint_event = self.base.make_event_from_transaction(
+                transaction=context.transaction,
+                tx_log=context.tx_log,
+                event_type=HistoryEventType.RECEIVE,
+                event_subtype=HistoryEventSubType.NONE,
+                asset=self.native_asset,
+                balance=Balance(amount=amount),
+                location_label=to_address,
+                notes=f'Mint {amount} {self.eure_token.symbol}',
+                counterparty=CPT_MONERIUM,
+            )
+            event = mint_event
+
+        elif (to_address == ZERO_ADDRESS and
+                context.transaction.input_data.hex().startswith(BURN_MONERIUM_SIGNATURE)):
+            # Create a burn event
+            burn_event = self.base.make_event_from_transaction(
+                transaction=context.transaction,
+                tx_log=context.tx_log,
+                event_type=HistoryEventType.SPEND,
+                event_subtype=HistoryEventSubType.NONE,
+                asset=self.native_asset,
+                balance=Balance(amount=amount),
+                location_label=from_address,
+                notes=f'Burn {amount} {self.eure_token.symbol}',
+                counterparty=CPT_MONERIUM,
+            )
+            event = burn_event
+
+        return DecodingOutput(event=event, refresh_balances=False)
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        # Not iterating over the events because the logic for decoding erc_20 transfers
+        # has not been executed before calling this decoding logic
```

Comment:
Move the comment to where the if statements are. Here it doesn't make much sense